apps/transport/lib/jobs/expiration_notification_job.ex

```diff
@@ -0,0 +1,226 @@
+defmodule Transport.Jobs.ExpirationNotificationJob do
+  @moduledoc """
+  This job sends daily digests to reusers about the expiration of their favorited datasets.
+  The expiration delays is the same for all reusers and cannot be customized for now.
+
+  It has 2 `perform/1` methods:
+  - a dispatcher one in charge of identifying contacts we should get in touch with today
+  - another in charge of building the daily digest for a specific contact (with only their favorited datasets)
+  """
+  use Oban.Worker,
+    max_attempts: 3,
+    tags: ["notifications"],
+    # Make sure we don't send twice the daily digest to a contact.
+    # The relevant `perform/1` uses (contact_id + digest_date) as args
+    # to leverage the unique clause.
+    unique: [period: {20, :hours}, fields: [:args, :queue, :worker]]
+
+  @type delay() :: integer()
+  @type dataset_ids() :: [integer()]
+  @type datasets() :: [DB.Dataset.t()]
+
+  import Ecto.Query
+  @notification_reason DB.NotificationSubscription.reason(:expiration)
+  # If delay < 0, the resource is already expired
+  @default_outdated_data_delays [-30, -7, 0, 7, 14]
+
+  @impl Oban.Worker
+  def perform(%Oban.Job{args: %{"contact_id" => contact_id, "digest_date" => digest_date}}) do
+    contact = DB.Repo.get!(DB.Contact, contact_id)
+    subscribed_dataset_ids = subscribed_dataset_ids_for_expiration(contact)
+
+    filtered_expiration_data =
+      digest_date
+      |> Date.from_iso8601!()
```

Comment:
On ne peut pas passer un `Date` directement dans le payload du job ?

Reply:
On passe bien une `%Date{}` quand on enqueue le job. Toutefois les arguments sont sérialisés en JSON, on récupère donc une chaine de caractères dans le job.

[Oban doc](https://hexdocs.pm/oban/Oban.html#module-defining-workers)
> Note that the args field of the job struct will always have string keys, regardless of the key type when the job was enqueued. The args are stored as JSON and the serialization process automatically stringifies all keys. Also, because args are always encoded as JSON, you must ensure that all values are serializable, otherwise you'll have encoding errors when inserting jobs.

Reply:
👍